mono/mini/mini-exceptions.c

```diff
@@ -2787,8 +2787,14 @@ mono_setup_altstack (MonoJitTlsData *tls)
 		if (mono_mprotect (tls->stack_ovf_guard_base, tls->stack_ovf_guard_size, MONO_MMAP_NONE)) {
 			/* mprotect can fail for the main thread stack */
 			gpointer gaddr = mono_valloc (tls->stack_ovf_guard_base, tls->stack_ovf_guard_size, MONO_MMAP_NONE|MONO_MMAP_PRIVATE|MONO_MMAP_ANON|MONO_MMAP_FIXED, MONO_MEM_ACCOUNT_EXCEPTIONS);
-			g_assert (gaddr == tls->stack_ovf_guard_base);
-			tls->stack_ovf_valloced = TRUE;
+			if (gaddr) {
+				g_assert (gaddr == tls->stack_ovf_guard_base);
+				tls->stack_ovf_valloced = TRUE;
+			} else {
+				/* couldn't allocate guard page, continue without it */
+				tls->stack_ovf_guard_base = NULL;
+				tls->stack_ovf_guard_size = 0;
```

Comment:
true, I changed it so that we call `mono_gc_register_altstack` if `tls->stack_ovf_guard_base` is set. `mono_gc_register_altstack` is a boehm thing btw.

Reply:
Yeah. What I meant is that the computation of the stack start/size was not correct. (instead of being guard_addr + guard_size it is just staddr). The stack should be registered even if we didn't allocate the overflow page.

Reply:
fixed